compiler/rustc_metadata/src/creader.rs

```diff
@@ -985,13 +986,54 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
         }
     }
 
+    #[allow(rustc::untranslatable_diagnostic)] // FIXME: make this translatable
```

Comment:
Is there a reason to `allow` this instead of adding the lint message to `messages.ftl`, and naming that ID with the error emitter? I can accept this with a FIXME, but I would prefer that the FIXME be fully qualified with the reason it could not be translated currently.

It seems `buffer_lint` accepts a DiagMessage, so it's just a matter of moving the diagnostic into the Fluent file and naming that diagnostic's identifier string.

Reply:
> Is there a reason to `allow` this instead of adding the lint message to `messages.ftl`, and naming that ID with the error emitter?

There isn't. While trying to find out what this is I just stumbled on a lot of `allow`s in Rustc, so I just went with it.
Will look into it.

Reply:
Done.
Let me know if I got anything wrong here.